qualtran/bloqs/arithmetic/comparison.py

```diff
@@ -953,10 +954,61 @@ def wire_symbol(self, reg: Optional[Register], idx: Tuple[int, ...] = tuple()) -
             return TextBox(f"⨁(x = {self.val})")
         raise ValueError(f'Unknown register symbol {reg.name}')
 
+    def is_symbolic(self):
+        return is_symbolic(self.bitsize, self.val)
+
+    def build_composite_bloq(
+        self, bb: 'BloqBuilder', x: 'Soquet', target: 'Soquet'
+    ) -> Dict[str, 'SoquetT']:
+        if self.is_symbolic():
+            raise DecomposeTypeError(f"cannot decompose symbolic {self}")
+
+        bits_k = x.reg.dtype.to_bits(self.val)
+
+        if self.bitsize == 1:
+            if self.val == 0:
+                x = bb.add(XGate(), q=x)
+            x, target = bb.add(CNOT(), ctrl=x, target=target)
+            if self.val == 0:
+                x = bb.add(XGate(), q=x)
```

Comment:
can you add a note that this is just a negative-control CNOT so that in the future, if and when we add a `cv` attribute to `CNOT` we can replace it